src/snake_game/utils.cljs

```diff
@@ -1,74 +0,0 @@
-(ns snake-game.utils
-  (:require [reagent.core :as reagent :refer [atom]]))
-
-(def key-code->move
-  "Mapping from integer key code to direction vector corresponding to that key"
-  {38 [0 -1]
-   40 [0 1]
-   39 [1 0]
-   37 [-1 0]})
-
-(defn rand-free-position
-  "Fuction takes snake and board-size as arguments.,
-  and returns random position not colliding wit snake body"
-  [snake [x y]]
-  (let [snake-positions (into #{} (:body snake))
-        board-positions (for [x-pos (range x)
-                              y-pos (range y)]
-                          [x-pos y-pos])]
-    (when-let [free-positions (seq (remove snake-positions board-positions))]
-      (rand-nth free-positions))))
-
-(defn collisions
-  "Returns true if snake collision with board edges or itself (snake body) is detected"
-  [{:keys [snake board] :as db}]
-  (let [{:keys [body direction]} snake
-        [x y] board
-        border-x #{x -1}
-        border-y #{y -1}
-        future-x (+ (first direction) (first (first body)))
-        future-y (+ (second direction) (second (first body)))]
-    (or (contains? border-x future-x)
-        (contains? border-y future-y)
-        (contains? (into #{} (rest body)) [future-x future-y]))))
-
-(defn change-snake-direction
-  "Changes snake head direction, only when it's perpendicular to the old head direction"
-  [[new-x new-y] [x y]]
```

Comment:
In real project I tend to avoid destructuring in the params. I have been taught by my Clojure teacher when working on Counterclockwise that you impose a bigger restriction on the caller by doing this. Besides, if your implementation changes, say you want to pattern match inside the function, you have to change the signature.